Library/Formula/rust.rb

```diff
@@ -23,6 +42,34 @@ def install
     system "./configure", *args
     system "make"
     system "make install"
+
+    resource("cargo").stage do
+      cargo_stage_path = pwd
```

Comment:
No, `buildpath` refers to the formula's build path, not the resource stage, since [`FileUtils.mktemp`](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/extend/fileutils.rb#L9) does a `cd` (and popped back upon `ensure`). This is critical here because staged resources get nested, therefore capturing `pwd` in a parent "scope" is necessary, which is fine for ruby the language, but an ugly hack if you ask me in terms of Homebrew DSL. The following would look much better, doesn't carry state in the Resource instance, and trivially allows nesting:

```ruby
resource("cargo").stage do |_, stage_path|
   #...
end
```